anesthetic/plotting/_matplotlib/hist.py

```diff
@@ -180,18 +179,18 @@ def __init__(
     ) -> None:
         super().__init__(data, bins=bins, bottom=bottom, **kwargs)
 
-    def _calculate_bins(self, data):
-        if 'range' not in self.kwds or self.kwds['range'] is None:
+    def _calculate_bins(self, data, bins):
+        nd_values = data.infer_objects(copy=False)._get_numeric_data()
+        values = np.ravel(nd_values)
+        values = values[~isna(values)]
```

Comment:
I'm not happy with repeating this a third time, and I haven't checked if the weights definitely need checking for nans

Reply:
Well, considering that `Hist1dPlot` was created for a `WeightedSeries`, we could simplify things here. 
Even simpler than my previous suggestion:
```python
            xmin, xmax = data.quantile(q).to_numpy().ravel()
```

Not sure what is easier for maintenance in the long run, though, so happy with either approach.

Reply:
That's too tidy to resist!